plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hadoop/TrinoHadoopCatalog.java

```diff
@@ -220,7 +219,13 @@ public void renameTable(ConnectorSession session, SchemaTableName from, SchemaTa
     @Override
     public Table loadTable(ConnectorSession session, SchemaTableName schemaTableName)
     {
-        return getCatalog(session).loadTable(toTableId(schemaTableName));
+        try {
+            return getCatalog(session).loadTable(toTableId(schemaTableName));
+        }
+        catch (NoSuchTableException e) {
+            // Have to change exception types due to code relying on specific exception to be thrown.
+            throw new TableNotFoundException(schemaTableName);
```

Comment:
Should we still include the root cause of the Exception to make debugging easier / possible?
EDIT: Not sure why GitHub won't let me use a 'suggestion' here...
```java
throw new TableNotFoundException(schemaTableName, e);
```